tests/commands/thread_test.py

```diff
@@ -87,3 +91,78 @@ def test_real_name_is_never_considered(self):
         mine = 'alot team'
         actual = thread.ReplyCommand.clear_my_address(mine, expected)
         self.assertListEqual(actual, expected)
+
+
+class _AccountTestClass(Account):
+    """Implements stubs for ABC methods."""
+
+    def send_mail(self, mail):
+        pass
+
+
+class TestDetermineSender(unittest.TestCase):
+
+    header_priority = ["From", "To", "Cc", "Envelope-To", "X-Envelope-To",
+                       "Delivered-To"]
+    mailstring = '\n'.join([
+        "From: from@example.com",
+        "To: to@example.com",
+        "Cc: cc@example.com",
+        "Envelope-To: envelope-to@example.com",
+        "X-Envelope-To: x-envelope-to@example.com",
+        "Delivered-To: delivered-to@example.com",
+        "Subject: Alot test",
+        "\n",
+        "Some content",
+        ])
+    # ['{}: {}@example.com'.format(header, header.lower()) for header in header_priority]
+    mail = email.message_from_string(mailstring)
+
+    def _test(self, accounts, expected):
+        with mock.patch('alot.commands.thread.settings.get_accounts',
+                        mock.Mock(return_value=accounts)):
+            with mock.patch('alot.commands.thread.settings.get',
+                            mock.Mock(return_value=self.header_priority)):
+                actual = thread.determine_sender(self.mail)
+        self.assertTupleEqual(actual, expected)
+
+    def test_assert_that_some_accounts_are_defined(self):
+        with mock.patch('alot.commands.thread.settings.get_accounts',
+                        mock.Mock(return_value=[])) as cm1:
+            with self.assertRaises(AssertionError) as cm2:
+                thread.determine_sender(None)
+        expected = ('no accounts set!',)
+        cm1.assert_called_once_with()
+        self.assertTupleEqual(cm2.exception.args, expected)
```

Comment:
It is documented for the stdlib: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises
and the test passes so I assume it works.